src/main/kotlin/org/radarbase/push/integration/garmin/resource/GarminPushEndpoint.kt

```diff
@@ -20,86 +22,128 @@ import javax.ws.rs.core.Response
     fun addEpochSummaries(): Response {
-        return healthApiService.processEpochs(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processEpochs(tree, user)
+        })
     }
 
     @POST
     @Path("sleeps")
     fun addSleeps(): Response {
-        return healthApiService.processSleeps(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processSleeps(tree, user)
+        })
     }
 
     @POST
     @Path("bodyCompositions")
     fun addBodyCompositions(): Response {
-        return healthApiService.processBodyCompositions(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processBodyCompositions(tree, user)
+        })
     }
 
     @POST
     @Path("stress")
     fun addStress(): Response {
-        return healthApiService.processStress(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processStress(tree, user)
+        })
     }
 
     @POST
     @Path("userMetrics")
     fun addUserMetrics(): Response {
-        return healthApiService.processUserMetrics(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processUserMetrics(tree, user)
+        })
     }
 
     @POST
     @Path("moveIQ")
     fun addMoveIQ(): Response {
-        return healthApiService.processMoveIQ(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processMoveIQ(tree, user)
+        })
     }
 
     @POST
     @Path("pulseOx")
     fun addPluseOX(): Response {
-        return healthApiService.processPulseOx(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processPulseOx(tree, user)
+        })
     }
 
     @POST
     @Path("respiration")
     fun addRespiration(): Response {
-        return healthApiService.processRespiration(tree, user)
+        return processResponses(function = { tree: JsonNode, user: User ->
+            healthApiService.processRespiration(tree, user)
+        })
     }
 
-    @POST
-    @Path("deregister")
-    fun deregisterUser(): Response {
-        return healthApiService.handleDeregistration(user)
+    /**
+     * Processes responses for all users
+     * @param function: The function to use to process data
+     */
+    private fun processResponses(function: (JsonNode, User) -> Response): Response {
+        val responses = userTreeMap.map { (user, tree) ->
+            function(tree, user)
+        }
+        if (
+            authMetadata.getOrDefault("isAnyUnauthorised", "false").toBoolean()
+        ) {
+            throw HttpUnauthorizedException(
+                "invalid_auth", "One of the users did not have " +
+                    "correct authorisation information."
```

Comment:
Does this mean if `isAnyUnauthorised` becomes true it will still send a 401 after processing the filtered/authorised users' data?

Reply:
Yes exactly.